### PR TITLE
Issue / Swagger Schema Id

### PR DESCRIPTION
**Pull request by SeferMirza**
Reproduce and solve swagger schema id conflict. It occurs when two different
controllers have a nested class with the same name.

## Tasks

> The problem is, since the schema uses only the class name when creating the schema, there is a conflict when there is a different class with the same name.
- [x] fix swagger schema create confilict problem
  - give ~~full names~~ custom names when creating schema
 
## Additional Tasks

- [x] Add `Searcher` and `GetMe` to `Spec`
- [x] Switch to grouped controller template for generated samples
  - use `_serviceProvider` and find target per action
- [x] Add url and guid spec extensions

### CODE REVIEW

File: src/blueprints/Do.Blueprints.Service.Application/Greeting/Swagger/SwaggerGreetingFeature.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 using Do.Architecture;
 using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using MySqlX.XDevAPI.Common;
```
Comment (dncsvr) [inline src/blueprints/Do.Blueprints.Service.Application/Greeting/Swagger/SwaggerGreetingFeature.cs:5]: using'i kaldırabilirz kullanılmıyor

File: src/blueprints/Do.Blueprints.Service.Application/ServiceSpecExtensions.cs [hunk 2]
```diff
@@ -3,19 +3,26 @@
...
     public static T The<T>(this Stubber _, params object?[] mockOverrides) where T : notnull =>
         ServiceSpec.ServiceProvider.OverrideMocksAndGetRequiredService<T>(mockOverrides);
 
+    public static T An<T>(this Searcher getMe, params object?[] mockOverrides) where T : notnull =>
+        getMe.Spec.GiveMe.An<T>(mockOverrides);
     public static T An<T>(this Stubber _, params object?[] mockOverrides) where T : notnull =>
         ServiceSpec.ServiceProvider.OverrideMocksAndGetRequiredService<T>(mockOverrides);
 
+    public static T A<T>(this Searcher getMe, params object?[] mockOverrides) where T : notnull =>
+        getMe.Spec.GiveMe.A<T>(mockOverrides);
```
Comment (dncsvr) [inline src/blueprints/Do.Blueprints.Service.Application/ServiceSpecExtensions.cs:25]: Searcher'ı veri tabanına yazılmış bir entity'yi belirli kriterlere göre alalım , o yüzden `The<>` olmalı extension diye düşünmüştük, `A<>` veya `An<>` dersek herhangi bir manasına da gelebiliyor, zaten o zaman da mocklarız, gerek olmaz diye düşünüyoum

Comment (dncsvr) [reply src/blueprints/Do.Blueprints.Service.Application/ServiceSpecExtensions.cs]: `ServiceSpec` zaten elimizde var oradan ServiceProvider'a ulaşabiliriz, `GiveMe`'den almaya gerek yok

File: src/blueprints/Do.Blueprints.Service.Application/Greeting/Swagger/SwaggerGreetingFeature.cs [hunk 3]
```diff
@@ -16,5 +17,26 @@ public void Configure(LayerConfigurator configurator)
...
+                        .Replace($"{t.Namespace}.", "")
+                        .Replace("Controller", "")
+                        .Replace("+", ".")!;
+                }
+
+                return splitedNamespace.Length > 1
+                    ? $"{string.Join('.', splitedNamespace.Skip(1))}.{name}"
+                    : name;
+            });
+        });
```
Comment (cihandeniz) [inline src/blueprints/Do.Blueprints.Service.Application/Greeting/Swagger/SwaggerGreetingFeature.cs:40]: greeting is not correct feature to configure schema ids, we should have a better fit for this one. please discuss it with @mouseless/developers

File: src/blueprints/Do.Blueprints.Service.Application/ServiceSpecExtensions.cs [hunk 4]
```diff
@@ -3,13 +3,17 @@
...
 
 namespace Do;
 
 public static class ServiceSpecExtensions
 {
     #region MockOverrider
 
+    public static T The<T>(this Searcher _, params object?[] mockOverrides) where T : notnull =>
+        ServiceSpec.ServiceProvider.OverrideMocksAndGetRequiredService<T>(mockOverrides);
+
```
Comment (cihandeniz) [inline src/blueprints/Do.Blueprints.Service.Application/ServiceSpecExtensions.cs:16]: searcher is only for searching for existing entities, remove this extension

File: src/core/Do.Architecture/Testing/SpecExtensions.cs [hunk 5]
```diff
@@ -15,6 +15,7 @@ public static class SpecExtensions
 
     #region Object
 
+    public static T AnInstanceOf<T>(this Searcher getMe) => getMe.Spec.GiveMe.AnInstanceOf<T>();
```
Comment (cihandeniz) [inline src/core/Do.Architecture/Testing/SpecExtensions.cs:18]: remove this

File: unreleased.md [hunk 6]
```diff
@@ -1 +1,14 @@
 # Unreleased
+
+## Features
+
+- `Documentation` feature is added with `Default` implementation.
```
Comment (cihandeniz) [inline unreleased.md:5]: indicate feature is beta like in previous release notes

**Review by cihandeniz — APPROVED** [review]

**Pull request merged**